packages/utils/src/composeAbortSignals.ts

```diff
@@ -0,0 +1,35 @@
+function findFirstPreAbortedSignal(signals: Iterable<AbortSignal>): AbortSignal | undefined {
+    for (const signal of signals) {
+        if (signal.aborted) {
+            return signal
+        }
+    }
+    return undefined
+}
+
+/**
+ * Compose a single AbortSignal from multiple AbortSignals with "OR" logic.
+ */
+export function composeAbortSignals(...signals: AbortSignal[]): AbortSignal {
+    if (signals.length === 0) {
+        throw new Error('must provide at least one AbortSignal')
+    }
+
+    const preAbortedSignal = findFirstPreAbortedSignal(signals)
```

Comment:
```suggestion
    const preAbortedSignal = signals.find((s) => (s.aborted === true))
```

We can then remove the `findFirstPreAbortedSignal` method

Reply:
Fixed 👍